Gems/Terrain/Code/Source/TerrainRenderer/TerrainClipmapManager.h

```diff
@@ -22,35 +24,70 @@ namespace Terrain
         AZ_RTTI(TerrainClipmapManager, "{5892AEE6-F3FA-4DFC-BBEC-77E1B91506A2}");
         AZ_DISABLE_COPY_MOVE(TerrainClipmapManager);
 
-        TerrainClipmapManager() = default;
+        TerrainClipmapManager();
         virtual ~TerrainClipmapManager() = default;
 
+        enum ClipmapName : uint32_t
+        {
+            MacroColor = 0,
+            MacroNormal,
+            DetailColor,
+            DetailNormal,
+            DetailHeight,
+            // Miscellany clipmap combining:
+            // roughness, specularF0, metalness, occlusion
+            DetailMisc,
+
+            Count
+        };
+
+        // Shader input names
+        static constexpr const char* ClipmapDataShaderInput = "m_clipmapData";
+        static constexpr const char* ClipmapImageShaderInput[ClipmapName::Count] = {
+            "m_macroColorClipmaps",
+            "m_macroNormalClipmaps",
+            "m_detailColorClipmaps",
+            "m_detailNormalClipmaps",
+            "m_detailHeightClipmaps",
+            "m_detailMiscClipmaps"
+        };
+
         void Initialize(AZ::Data::Instance<AZ::RPI::ShaderResourceGroup>& terrainSrg);
         bool IsInitialized() const;
         void Reset();
         bool UpdateSrgIndices(AZ::Data::Instance<AZ::RPI::ShaderResourceGroup>& srg);
 
         void Update(const AZ::Vector3& cameraPosition, AZ::Data::Instance<AZ::RPI::ShaderResourceGroup>& terrainSrg);
 
-        AZ::Data::Instance<AZ::RPI::AttachmentImage> GetClipmapImage(const AZ::Name& clipmapName) const;
+        // Import the clipmap to the frame graph and set scope attachment access,
+        // so that the compute pass can build dependencies accordingly.
+        void ImportClipmap(ClipmapName clipmapName, AZ::RHI::FrameGraphAttachmentInterface attachmentDatabase) const;
+        void UseClipmap(ClipmapName clipmapName, AZ::RHI::ScopeAttachmentAccess access, AZ::RHI::FrameGraphInterface frameGraph) const;
+
+        AZ::Data::Instance<AZ::RPI::AttachmentImage> GetClipmapImage(ClipmapName clipmapName) const;
     private:
         void UpdateClipmapData(const AZ::Vector3& cameraPosition);
         void InitializeClipmapData();
         void InitializeClipmapImages();
 
-        static constexpr uint32_t ClipmapStackSize = 5;
+        static constexpr uint32_t MacroClipmapStackSize = 3;
+        static constexpr uint32_t DetailClipmapStackSize = 5;
         static constexpr uint32_t ClipmapSizeWidth = 1024;
         static constexpr uint32_t ClipmapSizeHeight = 1024;
 
+        static_assert(DetailClipmapStackSize >= MacroClipmapStackSize,
+            "Macro clipmaps use lower resolutions. Array construction will use"
+            "max(DetailClipmapStackSize, MacroClipmapStackSize).");
```

Comment:
Not blocking this review, but this doesn't really makes sense. It should be fine either way, generally I'd expect a user to have about the same number of macro clipmaps vs detail clipmaps. But it's easy to think of an example where there are more macro materials - For instance, if the highest res clipmap is 8m, detail materials extend to 256m, and they want to render 32km, you'd have:
- 8, 16, 32, 64, 128, 256 = 6 detail clipmaps
- 512, 1024, 2048, 4096, 8192, 16384, 32768 = 7 macro clipmaps

Reply:
I did in this way is because we are having a fixed max render size for both clipmaps. I'll make a change to other part of the code to reflect it in the next PR.